crates/primitives/src/hardfork.rs

```diff
@@ -161,72 +77,34 @@ impl Default for Hardfork {
     }
 }
 
-impl From<BlockNumber> for Hardfork {
-    fn from(num: BlockNumber) -> Hardfork {
-        match num {
-            _i if num < 1_150_000 => Hardfork::Frontier,
-            _i if num < 1_920_000 => Hardfork::Dao,
-            _i if num < 2_463_000 => Hardfork::Homestead,
-            _i if num < 2_675_000 => Hardfork::Tangerine,
-            _i if num < 4_370_000 => Hardfork::SpuriousDragon,
-            _i if num < 7_280_000 => Hardfork::Byzantium,
-            _i if num < 9_069_000 => Hardfork::Constantinople,
-            _i if num < 9_200_000 => Hardfork::Istanbul,
-            _i if num < 12_244_000 => Hardfork::Muirglacier,
-            _i if num < 12_965_000 => Hardfork::Berlin,
-            _i if num < 13_773_000 => Hardfork::London,
-            _i if num < 15_050_000 => Hardfork::ArrowGlacier,
-
-            _ => Hardfork::Latest,
+impl From<(&ChainSpec, BlockNumber)> for Hardfork {
```

Comment:
Unsure if this makes sense, I think just taking block numbers where applicable is better

Reply:
Yeah I did that because `Hardfork` had `From<BlockNumber>` previously, but it isn't used at all now.